libs/core/cornflow_core/models/user.py

```diff
@@ -0,0 +1,171 @@
+"""
+
+"""
+import random
+import string
+
+from cornflow_core.exceptions import InvalidCredentials
+from cornflow_core.shared import (
+    password_crypt,
+    database,
+    check_password_pattern,
+    check_email_pattern,
+)
+from .meta_models import TraceAttributesModel
+
+
+class UserBaseModel(TraceAttributesModel):
+    __abstract__ = True
+    id = database.Column(database.Integer, primary_key=True)
+    first_name = database.Column(database.String(128), nullable=True)
+    last_name = database.Column(database.String(128), nullable=True)
+    username = database.Column(database.String(128), nullable=False, unique=True)
+    password = database.Column(database.String(128), nullable=True)
+    email = database.Column(database.String(128), nullable=False, unique=True)
+
+    def __init__(self, data):
+        """
+
+        :param data:
+        :type data:
+        """
+        super().__init__()
+        self.first_name = data.get("first_name")
+        self.last_name = data.get("last_name")
+        self.username = data.get("username")
+        # TODO: handle better None passwords that can be found when using ldap
+        check_pass, msg = check_password_pattern(data.get("password"))
+        if check_pass:
+            self.password = self.__generate_hash(data.get("password"))
+        else:
+            raise InvalidCredentials(msg)
+
+        check_email, msg = check_email_pattern(data.get("email"))
+        if check_email:
+            self.email = data.get("email")
+        else:
+            raise InvalidCredentials(msg)
+
+    def update(self, data):
+        """
+        Updates the user information in the database
+
+        :param dict data: the data to update the user
+        """
+        # First we create the hash of the new password and then we update the object
+        new_password = data.get("password")
+        if new_password:
+            new_password = self.__generate_hash(new_password)
+            data["password"] = new_password
+        super().update(data)
```

Comment:
Maybe you should also check the password pattern here ?

Reply:
Now it get checked on the endpoint logic, not on the user model if I remember corrrectly